File: app/src/main/java/fr/narwhals/go/domain/Liberty.java
package fr.narwhals.go.domain;


public class Liberty extends Section {
	private static final long serialVersionUID = -8409620248088827028L;
	
	private Territory territory;

	public Liberty(SColor color, Point point, Goban goban) {
		super(color, point, goban);
		territory = null;
	}
	
	public Territory getTerritory() {
		return territory;
	}
	
	public boolean hasTerritory() {
		return territory != null;
	}
	
	public void setTerritory(Territory territory) {
		this.territory = territory;
	}
}

File: app/src/main/java/fr/narwhals/go/domain/Liberty.kt
package fr.narwhals.go.domain


class Liberty(color: Section.SColor, point: Point, goban: Goban) : Section(color, point, goban) {

    var territory: Territory? = null

    init {
        territory = null
    }

    fun hasTerritory(): Boolean {
        return territory != null
    }
}
